modules/ztp-checking-du-cluster-config.adoc

```diff
@@ -0,0 +1,518 @@
+
+.Procedure
+
+. Check that the default Operator Hub sources are disabled. Run the following command:
++
+[source,terminal]
+----
+$ oc get operatorhub cluster -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+spec:
+    disableAllDefaultSources: true
+----
+
+. Check that all required `CatalogSource` resources are annotated for workload partitioning (`PreferredDuringScheduling`), for example:
++
+[source,terminal]
+----
+$ oc get catalogsource -A -o jsonpath='{range .items[*]}{.metadata.name}{" -- "}{.metadata.annotations.target\.workload\.openshift\.io/management}{"\n"}{end}'
+----
++
+.Example output
+[source,terminal]
+----
+certified-operators -- {"effect": "PreferredDuringScheduling"}
+community-operators -- {"effect": "PreferredDuringScheduling"}
+ran-operators -- <1>
+redhat-marketplace -- {"effect": "PreferredDuringScheduling"}
+redhat-operators -- {"effect": "PreferredDuringScheduling"}
+----
+<1> `CatalogSource` resources that are not annotated are also returned. In this example, the `ran-operators` `CatalogSource` resource is not annotated and does not have the `PreferredDuringScheduling` annotation.
++
+[NOTE]
+====
+In a properly configured vDU cluster, only a single annotated catalog source is listed.
+====
+
+. Check that all applicable {product-title} Operator namespaces are annotated for workload partitioning. This includes all Operators installed with core {product-title} and the set of additional Operators included in the reference DU tuning configuration.
++
+[source,terminal]
+----
+$ oc get namespaces -A -o jsonpath='{range .items[*]}{.metadata.name}{" -- "}{.metadata.annotations.workload\.openshift\.io/allowed}{"\n"}{end}'
+----
++
+.Example output
+[source,terminal]
+----
+default --
+openshift-apiserver -- management
+openshift-apiserver-operator -- management
+openshift-authentication -- management
+openshift-authentication-operator -- management
+----
++
+[IMPORTANT]
+====
+Additional Operators must not be annotated for workload partitioning. In the output from the previous command, additional Operators should be listed without any value on the right-hand side of the `--` separator.
+====
+
+. Check that the `ClusterLogging` configuration is correct.
+
+.. Validate that the curation schedule is appropriate for your application by running the following command:
++
+[source,terminal]
+----
+$ oc get -n openshift-logging ClusterLogging
+----
++
+.Example output
+[source,yaml]
+----
+apiVersion: logging.openshift.io/v1
+kind: ClusterLogging
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  managementState: "Managed"
+  curation:
+    type: "curator"
+    curator:
+      schedule: "30 3 * * *"
+  collection:
+    logs:
+      type: "fluentd"
+      fluentd: {}
+----
+
+.. Validate that the appropriate input logs and destination outputs are configured:
++
+[source,terminal]
+----
+$ oc get -n openshift-logging ClusterLogForwarder
+----
++
+.Example output
```

Comment:
Above cmd will not generate following output. 
1. the cmd only list the CRs, without giving any details.
2. After a CR is applied, usually the output would include extra fields, as well as status of the CR. 

This is an example of input/output. But I'm not sure if the CR will always be named "instance".
```
$ oc get -n openshift-logging clusterlogforwarders.logging.openshift.io instance -o yaml
apiVersion: logging.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  creationTimestamp: "2022-07-07T18:22:56Z"
  generation: 1
  name: instance
  namespace: openshift-logging
  resourceVersion: "5426172"
  uid: c54888f3-f75f-4534-aaf2-ccc1a1c7ddc6
spec:
  inputs:
  - infrastructure: {}
    name: infra-logs
  outputs:
  - name: kafka-open
    type: kafka
    url: tcp://10.46.55.190:9092/test
  pipelines:
  - inputRefs:
    - audit
    name: audit-logs
    outputRefs:
    - kafka-open
  - inputRefs:
    - infrastructure
    name: infrastructure-logs
    outputRefs:
    - kafka-open
status:
  conditions:
  - lastTransitionTime: "2022-07-18T14:37:32Z"
    status: "True"
    type: Ready
  inputs:
    audit:
    - lastTransitionTime: "2022-07-18T14:37:32Z"
      status: "True"
      type: Ready
    infrastructure:
    - lastTransitionTime: "2022-07-18T14:37:32Z"
      status: "True"
      type: Ready
  outputs:
    kafka-open:
    - lastTransitionTime: "2022-07-18T14:37:32Z"
      status: "True"
      type: Ready
  pipelines:
    audit-logs:
    - lastTransitionTime: "2022-07-18T14:37:32Z"
      status: "True"
      type: Ready
    infrastructure-logs:
    - lastTransitionTime: "2022-07-18T14:37:32Z"
      status: "True"
      type: Ready


```

Reply:
Can we just do `oc get -n openshift-logging clusterlogforwarders.logging.openshift.io -o yaml`?

Reply:
yes we can, but the output will be different.

Reply:
OK. I just diffed the two. This is the only difference:
```
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

Reply:
It should also indent the instance's content to be an element in that List. I think we can go with `oc get -n openshift-logging ClusterLogForwarder instance`. To be something other than `instance` the user would have to explicitly update the name. That is something they should know and can adjust the command accordingly.

Reply:
Ok I'll update as instance.